tests/pos/overloaded.scala

```diff
@@ -57,3 +57,30 @@ object overloaded {
   val a: A = fr(new C)
   val b: B = fr(new C)
 }
+
+// from #1381
+
+object Foo {
+  class Bar[T]
+  implicit def const[T](x: T): Bar[T] = ???
+
+  def bar[T](e: T): Any = ???
+  def bar[T](e: Bar[T]): Any = ???
+
+  val b: Bar[Int] = ???
+  bar(b)
+}
+
+object Test2 {
+    trait A; trait B
+    class C1 {
+       def f(x: A): Unit = println("A")
+    }
+    class C2 extends C1 {
+       def f(x: B): Unit = println("B")
+    }
+    object Test extends C2 with App {
```

Comment:
These tests should probably be in `run` to check if the correct overload of `f` is called.